### ISSUE TITLE
Maple devices continue last operation (beeping, rumbling, rendering, etc) after driver shuts down

### ISSUE THREAD

**Issue by gyrovorbis** [KallistiOS/KallistiOS#229]
This is a very minor issue that I can get around to eventually, but it's going to be pretty annoying... We already have examples of beeping and rendering to the VMU screens that show off this issue...

If you begin playing a tone on the VMU or if you render some text or presumably if you cause the puru pack to rumble, then exit an example, it will continue playing indefinitely. That is until you power cycle the DC, load another example, or physically take the peripheral out and put it back in.

Basically before the maple peripheral drivers shut down, they need to be sending stop commands to the devices for these kinds of operations; however, if you look at the top level `maple_shutdown()` driver callback, notice it's actually shutting down the maple hardware before even calling shutdown on each peripheral, so this will either have to be swapped or addressed in another manner.

**Comment by QuzarDC** [comment]
Switching the ordering of the peripheral shutdowns should be simple, and is certainly more correct. A big additional problem is that none of the cleanup actions that would need to be taken could happen within an interrupt (sending commands to the vmu or rumble). Since the most common way to exit is arch_exit in the cont_buttons callback, this could be fixed with #209 or would have to have the basic behaviors of the drivers otherwise changed (come up with an int-safe send method just to send the stop).

It could also be that there's some missing low-level signal that should be sent to have all devices cease operation and sending individualized ones isn't the 'correct' way to do it.

**Comment by QuzarDC** [comment]
To not create a lot of needless traffic I was pondering how to only send if a beep/rumble were still in progress on shutdown. It doesn't seem like there's any way to read the current status on those, but hit on the fact that there's a reset command intended to reset the devices back to default which is what we'd want to do on shutdown anyways. (and I just realized I had the hint of being the right thing over a year ago in my last comment here)

Opened #772 to add a facility for sending that reset command to a device, and set it to happen at shutdown. Only problem is that it doesn't seem to have an impact on vmu_beep.
